lib/features.ts

```diff
@@ -95,3 +96,13 @@ export function isIOS () {
     // @ts-expect-error - MSStream is not in `window` type, unsurprisingly
     (/iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream);
 }
+
+export function isMobileSafari18 () {
+  if (isBrowser() && typeof navigator !== 'undefined' && typeof navigator.userAgent !== 'undefined') {
+    // eslint-disable-next-line new-cap
+    const { browser, os } = UAParser(navigator.userAgent);
```

Comment:
Does `UAParser` not have these kinds of safety checks? If you wrote it this way because  to `navigator.userAgent` "might be undefined", let UAParser find the user-agent string? The `new` will also avoid the need for the eslint annotation

```
    const { browser, os } = new UAParser().getResult();
```

I think your implementation/invocation is useful when you're running it in a node context or analyzing data.